lsp/README.md

```diff
@@ -0,0 +1,38 @@
+# AWS Documents Language Server
+
+Language server for AWS File types
+
+## Functionality
+
+This Language Server currently supports completion, hover, diagnostics for the following file types:
+- Buildspec
+
+## Structure
+
+```
+.
+── src
+    └── client
+        └── vscode/ - Minimal vscode extension to test the language server
+    └── server
+        └── server.ts - Language Server entry point
+        └── registry.ts - Registry of items
+        └── service.ts - Interfaces for Language Service and implementation of Backend Services that we call into (json language service, yaml language service)
+        └── utils/ - Utilities for general functionality
+        └── filetypes/ - Folder where filetype implementations live
+── test
+    └── unit/ - Unit tests
+        └── ... - These directories mirror src
+    └── integration/ - Integration tests
+        └── ... - These directories mirror src
```

Comment:
I know that the current structure is conventional for JS projects, however, I really like the idea of having test files live beside the implementation instead.